operator/pkg/metrics/exporter.go

```diff
@@ -0,0 +1,39 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metrics
+
+import (
+	"fmt"
+
+	ocprom "contrib.go.opencensus.io/exporter/prometheus"
+	"github.com/prometheus/client_golang/prometheus"
+	"go.opencensus.io/stats/view"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+)
+
+// RegisterOperatorMetricsServer registers HTTP endpoint to serve
+// custom operator metrics.This is different endpoint that the one
+// provided by controller-runtime (which contains metrics like reconcile
+// time, request latency and so on)
```

Comment:
should we combine actually?

Reply:
Combine to? A single endpoint `/metrics`? Not sure if that is possible
I found this somewhere deep inside k8s-controller-runtime package. They are doing what we are doing here. The only way I see to serve metrics at the same address is with `AddMetricsExtraHandler`. 
```go
func (cm *controllerManager) serveMetrics(stop <-chan struct{}) {
	handler := promhttp.HandlerFor(metrics.Registry, promhttp.HandlerOpts{
		ErrorHandling: promhttp.HTTPErrorOnError,
	})
	// TODO(JoelSpeed): Use existing Kubernetes machinery for serving metrics
	mux := http.NewServeMux()
	mux.Handle(defaultMetricsEndpoint, handler)

	func() {
		cm.mu.Lock()
		defer cm.mu.Unlock()

		for path, extraHandler := range cm.metricsExtraHandlers {
			mux.Handle(path, extraHandler)
		}
	}()
```

Reply:
We can try registering at `/metrics` instead of op-metrics. But doesn't that mask the metrics provided by kubernetes controller runtime framework? It has stuffs we need like latency for reconciliation etc

Reply:
Digging into `controller-runtime`'s `pkg/metrics/client_go_adapter.go` I found out that they are using Prometheus API directly. However, the registry is exported. Let me see if I can use it.

Reply:
Done. It is unified. All metrics are available under `/metrics`.